src/main/java/com/hubspot/jinjava/lib/filter/SliceFilter.java

```diff
@@ -47,7 +51,30 @@ public Object filter(Object var, JinjavaInterpreter interpreter, String... args)
     }
 
     int slices = NumberUtils.toInt(args[0], 3);
-    return Iterators.paddedPartition(loop, slices);
+    if (slices <= 0) {
+      throw new InvalidArgumentException(interpreter, this, InvalidReason.POSITIVE_NUMBER, 0, args[0]);
+    }
+    List<List<Object>> result = new ArrayList<>();
+
+    List<Object> currentList = null;
+    int i = 0;
+    while (loop.hasNext()) {
+      Object next = loop.next();
+      if (i % slices == 0) {
+        currentList = new ArrayList<>(slices);
+        result.add(currentList);
+      }
+      currentList.add(next);
+      i++;
+    }
+
+    if (args.length > 1 && currentList != null) {
+      while (currentList.size() < slices) {
+        currentList.add(args[1]);
```

Comment:
nit: assign `args[1]` to something like `fill_with` for clarity

Reply:
Done